evalml/automl/automl_search.py

```diff
@@ -185,11 +181,6 @@ def __init__(self,
         else:
             raise TypeError("max_time must be a float, int, or string. Received a {}.".format(type(max_time)))
 
-        if max_pipelines:
-            if not max_iterations:
-                max_iterations = max_pipelines
-            logger.warning("`max_pipelines` will be deprecated in the next release. Use `max_iterations` instead.")
```

Comment:
Food for future thought: I do wonder if we should have used python `warnings` here instead of / in addition to our logger

Reply:
Yeah, I agree, could have been useful to do it in addition to logging so that it's much more visible to the user.

Reply:
What's the benefit of doing both? With the way the logger is defined now, logger.warning shows up in stdout.

Reply:
I personally think the logger is too easy to miss compared to the python warning:
![image](https://user-images.githubusercontent.com/5422235/95343367-70709b80-0886-11eb-962e-73eb8a5fee3b.png)

![image](https://user-images.githubusercontent.com/5422235/95343269-533bcd00-0886-11eb-826d-8425b861fe3b.png)